streaming/receiver.c

```diff
@@ -389,6 +527,16 @@ static int rrdpush_receive(struct receiver_state *rpt)
     info("STREAM %s [receive from [%s]:%s]: initializing communication...", rpt->host->hostname, rpt->client_ip, rpt->client_port);
     char initial_response[HTTP_HEADER_SIZE];
     if (rpt->stream_version > 1) {
+        if(rpt->stream_version >= STREAM_VERSION_COMPRESSION){
+#ifdef ENABLE_COMPRESSION
+            if(!rpt->rrdpush_compression)
+                rpt->stream_version = (STREAM_VERSION_COMPRESSION - 1);
```

Comment:
I suggest we use direct `STREAM_VERSION_CLABELS` instead to subtract the version.

Reply:
Done.